src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -122,11 +123,29 @@ private void processDirectors() {
                 if (!(tnt.getVelocity().lengthSquared() > 0.35) || tracking.containsKey(tnt)) {
                     continue;
                 }
-                Craft c = CraftManager.getInstance().fastNearestCraftToLoc(tnt.getLocation());
+
+                Craft c;
+
+                if(Config.EnableTNTTracking) {
+                    if(!tnt.hasMetadata("MCC-Sender")) {
+                        continue;
+                    }
+                    Player sender = Bukkit.getPlayer(UUID.fromString(tnt.getMetadata("MCC-Sender").get(0).asString()));
+                    if (sender == null) {
+                        continue;
+                    }
+
+                    c = CraftManager.getInstance().getCraftByPlayer(sender);
```

Comment:
How does this function respond to subcrafts, sunken crafts, and squadron crafts?  I suspect it may not function as expected.